tests/unit-tests/gpartition/test_hard_negative_post_partition.py

```diff
@@ -0,0 +1,278 @@
+"""
+    Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+import os
+import json
+import torch as th
+import numpy as np
+from typing import Dict
+
+import pytest
+
+from numpy.testing import assert_almost_equal
+from graphstorm.model.utils import load_dist_nid_map
+from dgl.data.utils import load_tensors, save_tensors
+from graphstorm.gpartition.post_hard_negative import (shuffle_hard_negative_nids,
+                                                      load_hard_negative_config)
+
+@pytest.fixture(scope="module", name="gsprocessing_hard_negative_config")
+def gsprocessing_config_hard_negative_dict_fixture() -> Dict:
```

Comment:
Cloud we save this to a file instead and use the fixture to load the JSON into a dict?

Reply:
Update to save into the file, as we need to test with loading from files.